modules/src/core/ics02_client/handler.rs

```diff
@@ -17,13 +19,17 @@ pub enum ClientResult {
 }
 
 /// General entry point for processing any message related to ICS2 (client functions) protocols.
-pub fn dispatch<Ctx>(ctx: &Ctx, msg: ClientMsg) -> Result<HandlerOutput<ClientResult>, Error>
+pub fn dispatch<Ctx>(
+    now: Time,
+    ctx: &Ctx,
+    msg: ClientMsg,
+) -> Result<HandlerOutput<ClientResult>, Error>
 where
     Ctx: ClientReader,
 {
     match msg {
-        ClientMsg::CreateClient(msg) => create_client::process(ctx, msg),
-        ClientMsg::UpdateClient(msg) => update_client::process(ctx, msg),
+        ClientMsg::CreateClient(msg) => create_client::process(now.into(), ctx, msg),
+        ClientMsg::UpdateClient(msg) => update_client::process(now, ctx, msg),
```

Comment:
Would it make sense to harmonize the signatures of these `process` functions to use the same timestamp type, while you are at it?

Reply:
Fixed now.